### PR TITLE
SNOMED CT REST API improvements

### PR DESCRIPTION
**Pull request by cmark**
This PR adds quality of life improvements to the SNOMED CT REST API:
* Add `sort` query parameter to all search endpoints that have internal support for field-based sorting (`/concepts`, `/descriptions`, `/relationships`, `/refsets`, `/members`, `/classifications`, `/branches`).
* It is possible to specify the order for a given sortKey (`<field>:(asc|desc)`)
* Add `term` based sorting to `concepts` endpoint (`sort=term:desc`)
* Add `parent`, `ancestor`, `statedParent`, `statedAncestor` based filtering directly to `/concepts` endpoint
* Add POST HTTP method variants at `/search` endpoints for `/:path/concepts`, `/:path/descriptions`, `/:path/relationships`, `/:path/members`
* Fix styling of client and server errors section
* Fix nested SNOMED CT Query evaluations
* Fix `ascending` flag bug for script-based sorts

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchResourceRequest.java [hunk 1]
```diff
@@ -114,6 +123,10 @@ public boolean isAscending() {
 			return ascending;
 		}
 		
+		public static SortScript of(String script, final Map<String, Object> arguments, boolean isAscending) {
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchResourceRequest.java]: Please rename the third argument to `ascending`, and call the three-arg `of` factory method in the two-arg `of` methods below as well, for symmetry 🥂

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedConcept.java [hunk 2]
```diff
@@ -342,6 +371,7 @@ public SnomedConcepts getStatedDescendants() {
 		return statedParentIds == null ? null : Arrays.stream(statedParentIds).mapToObj(Long::toString).collect(Collectors.toList());
 	}
 	
+	@JsonIgnore
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedConcept.java:374]: I'm afraid of situations where someone sends in JSON with an updated enumeration value, and we don't act on it 😨 

Can/should we keep writeability while it is present in the JSON output?

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedDocument.java [hunk 3]
```diff
@@ -131,10 +132,10 @@ public B effectiveTime(final long effectiveTime) {
 	}
 	
 	public static class Fields extends RevisionDocument.Fields {
-		public static final String MODULE_ID = "moduleId";
-		public static final String RELEASED = "released";
-		public static final String ACTIVE = "active";
-		public static final String EFFECTIVE_TIME = "effectiveTime";
+		public static final String MODULE_ID = SnomedComponent.Fields.MODULE_ID;
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedDocument.java]: This introduces coupling between the domain and the indexed document world, which makes future changes to either of the classes harder (either one has to remember to rename properties in both classes, or rename in one and "disconnect" the constant value in the other)

**Review by apeteri — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedConcept.java (hunk 2 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedConcept.java:374]: We use special input types on the REST API side, that will prevent the scenario you are describing.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedDocument.java (hunk 3 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedDocument.java]: Then let's connect both to `SnomedRf2Headers` constants, where applicable.

**Review by apeteri — APPROVED** [review]

**Review by nagyo — APPROVED** [review]
LGTM!

**Pull request merged**